submissions/zalevskyi/dom/js/app.js

```diff
@@ -0,0 +1,110 @@
+        img_src: './img/Pingala.png',
+        img_alt: 'Pingala portrait'
+    },
+    'reyna' : {
+        type: 'governor',
+        title: 'The Financier',
+        description: 'Shrewd and fiscally knowledgeable, Reyna is able to strengthen the economic power of a city, using finances to provide more than is normally possible.',
+        context: 'A financier is an official tasked with resolving budgetary and financial matters, requiring an astute understanding of economics and the psychology of markets. Playing with the money of the state is not a matter for the faint-hearted.',
+        img_src: './img/Reyna.png',
+        img_alt: 'Reyna portrait'
+    },
+    'victor' : {
+        type: 'governor',
+        title: 'The Castellan',
+        description: 'Military focused, once established in a city, Victor is instrumental in strengthening, preserving, and defending it from foreign threats.',
+        context: 'A Castellan is a military officer responsible for directing the day-to-day affairs of a castle or fortress and the territory immediately around it. As castles were massive investments of treasure and important points of military power, the castellan had to be both competent and politically reliable. In times of war, a competent castellan could determine the fate of campaigns, and with that, the historical outcome.',
+        img_src: './img/Victor.png',
+        img_alt: 'Victor portrait'
+    }
+}
+const ARTICLE_INNER_HTML = {
+    'overview': `<h1 id='title' class='text_heading_1'></h1><p id='description' class='text'></p>`,
+    'governor': `<h1 id='title' class='text_heading_1'></h1>
+                 <img id='portrait' alt='' src='' class='portrait'>
+                 <h2 class='text_heading_2 description_heading'>Description</h2>
+                 <p id='description' class='text description'></p>
+                 <h2 class='text_heading_2 context_heading'>Historical Context</h2>
+                 <p id='context' class='text context'></p>`
+}
+Object.freeze(WIKI)
+Object.freeze(ARTICLE_INNER_HTML)
+function updateCommon(name) {
+    document.getElementById('title').textContent = `${name} - ${WIKI[name].title}`
+    document.getElementById('description').textContent = WIKI[name].description
+}
+function updateGovernor(name) {
+    document.getElementById('context').textContent = WIKI[name].context
+    document.getElementById('portrait').src = WIKI[name].img_src
+    document.getElementById('portrait').alt = WIKI[name].img_alt
+}
+document.getElementById('menu').addEventListener('click', ({target}) => {
+    const target_before = document.querySelector('.menu_item_selected')
```

Comment:
Why do you check for the previous clicked element? 
To my mind, it's a redundant operation. 
You can get rid of all checking of the previous elements and just switch content.

Reply:
Agree.
That was overthinking about performance improvement in extremely rare situation of selecting already selected item.